### PR TITLE
Implement `odgi` package

### PR DESCRIPTION
**Pull request by tbhaxor**
This commit re-implements odgi package (superseded by #38741)

### CODE REVIEW

File: var/spack/repos/builtin/packages/odgi/package.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    generator("make", default="make")
+
+    # the range is required to successfully build the program
+    requires("%gcc", msg="Package odgi depends on the gcc C++ compiler")
+    conflicts(
+        "%gcc@:9.2,13:", msg="Unsupported compiler version. Recommended range is 9.3 -> 12.x"
+    )
+
+    # <<< Dependencies list starts here
+    requires("%gcc@9.3:12", when="%gcc")
```
Comment (EbiArnie) [inline var/spack/repos/builtin/packages/odgi/package.py]: This is covered by the lines above. It should be removed.

**Review by EbiArnie — CHANGES_REQUESTED** [review]

**Review by tldahlgren — COMMENTED** [review]
Confirmed the home page and version commit.

File: var/spack/repos/builtin/packages/odgi/package.py (hunk 1 above)
Comment (tbhaxor) [reply var/spack/repos/builtin/packages/odgi/package.py]: Done!

**Review by EbiArnie — APPROVED** [review]
OK. I built the binary with this and ran 'odgi test', seems to work.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**